File: Array | Dictionary | Set/valid_anagram/py/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        dictionary = {}

        if len(s) != len(t):
            return False

        for i in range(len(s)):

            if not dictionary.get(s[i]):
                dictionary[s[i]] = 1
            else:
                dictionary[s[i]] = dictionary[s[i]] + 1 
        
            if not dictionary.get(t[i]):
                dictionary[t[i]] = -1
            else:
                dictionary[t[i]] = dictionary[t[i]] - 1
        
        for val in dictionary.values():
            if val != 0:
                return False
            
        return True

File: Array | Dictionary | Set/valid_anagram/ts/valid_anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length != t.length) {
        return false;
    }

    const dict = new Map();

    for ( let i = 0 ; i < s.length ; i++ ) {

        if ( ! dict.has(s[i]) ) {
            dict.set(s[i], 1)
        } else {
            dict.set(s[i], dict.get(s[i]) + 1) 
        }

        if ( ! dict.has(t[i]) ) {
            dict.set(t[i], -1)
        } else {
            dict.set(t[i], dict.get(t[i]) - 1) 
        }
    }

    for ( let [key,val] of dict ) {
        if (val != 0 ) {
            return false;
        }
    }

    return true;
};